src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = 'triangle';
 
-}
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
-export class Circle {
+  constructor(color: string, a: number, b: number, c: number) {
+    if (Math.min(a, b, c) <= 0) {
+      throw new Error('Triangle sides must be greater than 0');
+    }
 
+    if (a + b <= c || a + c <= b || b + c <= a) {
```

Comment:
I believe that you can do this  line more optimaze (you can create arr with sides and sort this arr after that you should only check one side)